mlflow/data/huggingface_dataset_source.py

```diff
@@ -0,0 +1,109 @@
+import datasets
+
+from typing import TypeVar, Any, Union, Optional, Mapping, Sequence, Dict
+
+from mlflow.data.dataset_source import DatasetSource
+
+
+HuggingFaceDatasetSourceType = TypeVar(
+    "HuggingFaceDatasetSourceType", bound="HuggingFaceDatasetSource"
+)
+
+
+class HuggingFaceDatasetSource(DatasetSource):
+    """
+    Represents the source of a Hugging Face dataset used in MLflow Tracking.
+    """
+
+    def __init__(
+        self,
+        path: str,
+        config_name: Optional[str] = None,
+        data_dir: Optional[str] = None,
+        data_files: Optional[
+            Union[str, Sequence[str], Mapping[str, Union[str, Sequence[str]]]]
+        ] = None,
+        split: Optional[Union[str, datasets.Split]] = None,
+        revision: Optional[Union[str, datasets.Version]] = None,
+        task: Optional[Union[str, datasets.TaskTemplate]] = None,
+    ):
+        """
+        :param path: The path of the Hugging Face dataset.
+        :param config_name: The name of of the Hugging Face dataset configuration.
```

Comment:
`datasets.load_dataset` calls this `name`, but it's very clearly documented as `Defining the name of the dataset configuration.` in https://huggingface.co/docs/datasets/v2.10.0/en/package_reference/loading_methods#datasets.load_dataset and `datasets.Dataset` objects themselves refer to this value as `ds.config_name`.